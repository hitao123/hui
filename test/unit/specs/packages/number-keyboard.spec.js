import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import numberKeyboard from '../../../../packages/number-keyboard';
import numberKeyboardDemo from '../../../../docs/src/views/numberkeyboard';

describe('Dialog component', () => {
  it('should show Dialog', () => {
    const warpper = shallowMount(numberKeyboard, {
      propsData: {
        closeButtonText: 'close'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show Button Demo', () => {
    expect(renderToString(numberKeyboardDemo)).toMatchSnapshot();
  });
});
