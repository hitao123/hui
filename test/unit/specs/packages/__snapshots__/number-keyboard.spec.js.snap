// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component should show Button Demo 1`] = `
<div class="demo-number">
  <h-password-input value=""></h-password-input>
  <h-number-keyboard closeButtonText="关闭"></h-number-keyboard>
</div>
`;

exports[`Dialog component should show Dialog 1`] = `
<div class="h-number-keyboard" style="display: none;" name="h-slide-bottom">
  <div class="h-hairline--top h-number-keyboard__title"><span></span> <span class="h-number-keyboard__close">close</span></div>
  <div class="h-number-keyboard__body">
    <key-stub text="1" type=""></key-stub>
    <key-stub text="2" type=""></key-stub>
    <key-stub text="3" type=""></key-stub>
    <key-stub text="4" type=""></key-stub>
    <key-stub text="5" type=""></key-stub>
    <key-stub text="6" type=""></key-stub>
    <key-stub text="7" type=""></key-stub>
    <key-stub text="8" type=""></key-stub>
    <key-stub text="9" type=""></key-stub>
    <key-stub text="." type="gray"></key-stub>
    <key-stub text="0" type=""></key-stub>
    <key-stub text="delete" type="gray,delete"></key-stub>
  </div>
</div>
`;
